gen/utils/data/class.go

```diff
@@ -178,20 +182,92 @@ func (c *Class) loadMetaFile() error {
 func (c *Class) setClassData() error {
 	genLogger.Debug(fmt.Sprintf("Setting class data for class '%s'.", c.ClassName))
 
-	c.setResourceName()
+	// TODO: add function to set AllowDelete
 
-	// TODO: add functions to set the other class data
+	// TODO: add function to set Children
+
+	// TODO: add function to set ContainedBy
+
+	// TODO: add placeholder function for Deprecated
+
+	// TODO: add placeholder function for DeprecatedVersions
+
+	// TODO: add function to set Documentation
+
+	// TODO: add function to set IdentifiedBy
+
+	// TODO: add function to set IsMigration
+
+	// TODO: add function to set IsRelational
+
+	// TODO: add function to set IsSingleNested
+
+	// TODO: add function to set PlatformType
 
 	if properties, ok := c.MetaFileContent["properties"]; ok {
 		c.setProperties(properties.(map[string]interface{}))
 	}
 
+	// TODO: add function to set RequiredAsChild
+
+	err := c.setResourceName()
+	if err != nil {
+		return err
+	}
+
+	err = c.setResourceNameNested()
+	if err != nil {
+		return err
+	}
+
+	// TODO: add function to set RnFormat
+
+	// TODO: add function to set Versions
+
 	genLogger.Debug(fmt.Sprintf("Successfully set class data for class '%s'.", c.ClassName))
 	return nil
 }
 
-func (c *Class) setResourceName() {
-	genLogger.Trace(fmt.Sprintf("Implement setting resourceName for class '%s'.", c.ClassName))
+func (c *Class) setResourceName() error {
+	genLogger.Debug(fmt.Sprintf("Setting resource name for class '%s'.", c.ClassName))
+
+	// TODO: add logic to override the resource name from an definition file.
```

Comment:
Not important but..
->
// TODO: add logic to override the resource name from a definition file.

Reply:
assuming the change was `an` to `a`, changed this now